From 6812fe3828d0ab6f1b857c40a5e11943a215411b Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Fri, 16 May 2014 23:34:54 +1000
Subject: [PATCH 1/3] Fix RemoteFile::GetRealFileSize() caching.

The timer was never started, and as such the function would always return the cached value
---
 mythtv/libs/libmythbase/remotefile.cpp |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythbase/remotefile.cpp b/mythtv/libs/libmythbase/remotefile.cpp
index 4f49ce38..4164eb0 100644
--- a/mythtv/libs/libmythbase/remotefile.cpp
+++ b/mythtv/libs/libmythbase/remotefile.cpp
@@ -993,7 +993,8 @@ long long RemoteFile::GetRealFileSize(void)
 
     QMutexLocker locker(&lock);
 
-    if (completed || lastSizeCheck.elapsed() < MAX_FILE_CHECK)
+    if (completed ||
+        (lastSizeCheck.isRunning() && lastSizeCheck.elapsed() < MAX_FILE_CHECK))
     {
         return filesize;
     }
-- 
1.7.10.2

